### PR TITLE
Add important details about external memory to docs

### PR DESCRIPTION
**Pull request by david-cortes**
Not 100% sure that I'm getting all the details right, so @trivialfis please take a careful look.

### CODE REVIEW

File: doc/tutorials/external_memory.rst [hunk 1]
```diff
@@ -26,9 +26,9 @@ The external memory support has undergone multiple development iterations. Like
 data batch-by-batch using a custom iterator supplied by the user. However, unlike the
 :py:class:`~xgboost.QuantileDMatrix`, external memory does not concatenate the batches
 (unless specified by the ``extmem_single_page``) . Instead, it caches all batches in the
-external memory and fetch them on-demand. Go to the end of the document to see a
-comparison between :py:class:`~xgboost.QuantileDMatrix` and the external memory version of
-:py:class:`~xgboost.ExtMemQuantileDMatrix`.
+external memory to disk (in a compressed format) and fetches them on-demand. Go to the
```
Comment (trivialfis) [inline doc/tutorials/external_memory.rst:29]: We use  `external memory` to refer to disk. For the CPU, the "main memory" is the memory we usually talk about, and external memory can refer to anything else from disk to network storage. For XGBoost, only disk is supported. As a result, in XGBoost "external memory" and "disk" are the same thing from the perspective of a CPU-based algorithm.

For GPU, the main memory is the device memory, whereas CPU memory and disk are "external" to GPU.

File: python-package/xgboost/core.py [hunk 2]
```diff
@@ -446,6 +446,9 @@ class DataIter(ABC):  # pylint: disable=too-many-instance-attributes
     cache_prefix :
         Prefix to the cache files, only used in external memory.
 
+        Note that using this class for non-quantile external memory **will cache data
+        on disk** under the path passed here.
```
Comment (trivialfis) [inline python-package/xgboost/core.py:450]:
```suggestion
        Note that using this class for external memory **will cache data
        on disk** under the path passed here.
```

File: doc/tutorials/external_memory.rst (hunk 1 above)
Comment (trivialfis) [reply doc/tutorials/external_memory.rst:29]: Maybe a brief intro with a few sentences. But for this PR, could you please help revise or revert the change?

Comment (david-cortes) [reply doc/tutorials/external_memory.rst]: Undid the changes to the .rst file.

**Review by trivialfis — APPROVED** [review]

**Pull request merged**